### ISSUE TITLE
Plan.retrieve with legacy Plan identifier are not escaped properly

### ISSUE THREAD

**Issue by jjlueck-tock** [stripe/stripe-java#736]
This is a bug in the 8.1.0 and 9.0.0 release of stripe-java.

We have a bunch of legacy subscription Plan objects in Stripe that were created before the ids are generated for us (using the `plan_xyz` convention). Many of these identifiers have forward slashes in them (e.g. `Pro plan $699/month`).

When we attempt to retrieve any of these plans, we get an InvalidRequestException.

```
com.stripe.exception.InvalidRequestException: Unrecognized request URL (GET: /v1/plans/Pro%20plan%20$699/month). Please see https://stripe.com/docs or we can help at https://support.stripe.com/.
	at com.stripe.net.LiveStripeResponseGetter.handleApiError(LiveStripeResponseGetter.java:742)
	at com.stripe.net.LiveStripeResponseGetter.staticRequest(LiveStripeResponseGetter.java:532)
	at com.stripe.net.LiveStripeResponseGetter.request(LiveStripeResponseGetter.java:88)
	at com.stripe.net.ApiResource.request(ApiResource.java:169)
	at com.stripe.model.Plan.retrieve(Plan.java:211)
	at com.stripe.model.Plan.retrieve(Plan.java:199)
```

Could we perhaps get the forward slashes URL escaped from within the library?

We did not have this invalid request problem with using the v7 version of this library.

**Comment by remi-stripe** [comment]
@jjlueck-tock Thanks a lot for the report. We're investigating and we'll update the library soon!

**Comment by mickjermsurawong-stripe** [comment]
Hi @jjlueck-tock. Sorry for the problem you have here. Will make a fix on this. Next release on V9 will have this patch.

**Comment by mickjermsurawong-stripe** [comment]
hi @jjlueck-tock, we have released 9.1 with the patch on id encoding. This release should have the same behavior for url encoding as version 7 now. Please let me know if you have other issues. Thank you for reporting this!
